src/library/tracksuggestion/tracksuggestionfeature.cpp

```diff
@@ -0,0 +1,423 @@
+    QString propertyMessage = tr("Update the title and artist if they are misspelled.");
+
+    QStringList items;
+
+    items << tr("Import metadata from MusicBrainz")
+          << tr("Update the track properties");
+
+    QString html;
+    html.append(QString("<h2>%1</h2>").arg(title));
+    html.append(QString("<p>%1</p>").arg(status));
+    html.append(QString("<p>%1</p>").arg(updateMessage));
+    html.append(QString("<p>%1</p>").arg(propertyMessage));
+
+    return html;
+}
+
+void TrackSuggestionFeature::activate() {
+    m_title = tr("Track Suggestion");
+    emit disableSearch();
+    emit enableCoverArtDisplay(false);
+    emit switchToView("TRACKSUGGESTION");
+}
+
+void TrackSuggestionFeature::activateChild(const QModelIndex& index) {
+    TreeItem* item = static_cast<TreeItem*>(index.internalPointer());
+    if (!(item && item->getData().isValid())) {
+        return;
+    }
+    qDebug() << "TrackSuggestionFeature item" << item->getLabel()
+             << "selected. The track ID is: " << item->getData();
+    TrackId trackId = TrackId(item->getData());
+    if (!trackId.isValid()) {
+        qDebug() << "Track ID is not valid";
+        return;
+    }
+    TrackPointer trackPointer = m_pLibrary->trackCollectionManager()->getTrackById(trackId);
+    emitTrackPropertiesToDialog(trackPointer);
+    m_pTrack = trackPointer;
+    if (!lookTrackHasSuggestions(trackId)) {
+        qDebug() << "Selected track has no suggestions fetched before.";
+        emit switchToView(kSuggestionTitle);
+    } else {
+        m_pSuggestionPlaylistModel->setPlaylist(trackId.toString());
+        emit showTrackModel(m_pSuggestionPlaylistModel);
+        emit enableCoverArtDisplay(false);
+    }
+}
+
+void TrackSuggestionFeature::slotTrackChanged(const QString& group,
+        TrackPointer pNewTrack,
+        TrackPointer pOldTrack) {
+    if (pNewTrack) {
+        playerInfoTrackLoaded(group, pNewTrack);
+    }
+}
+
+void TrackSuggestionFeature::playerInfoTrackLoaded(const QString& group, TrackPointer pNewTrack) {
+    QString artist = pNewTrack->getArtist();
+    QString title = pNewTrack->getTitle();
+    TrackId trackId = pNewTrack->getId();
+    if (!trackId.isValid()) {
+        qDebug() << "Track ID is not valid!";
+        return;
+    }
+    if (group == "[Channel1]") {
+        treeItemDeckOne->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckOne->setData(trackId.toVariant());
+    } else if (group == "[Channel2]") {
+        treeItemDeckTwo->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckTwo->setData(trackId.toVariant());
+    } else if (group == "[Channel3]") {
+        treeItemDeckThree->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckThree->setData(trackId.toVariant());
+    } else {
+        treeItemDeckFour->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckFour->setData(trackId.toVariant());
+    }
```

Comment:
```suggestion
    } else if (group == "[Channel3]") {
        treeItemDeckThree->setLabel(composeTreeItemLabel(artist, title));
        treeItemDeckThree->setData(trackId.toVariant());
    } else if (group == "[Channel4]") {
        treeItemDeckFour->setLabel(composeTreeItemLabel(artist, title));
        treeItemDeckFour->setData(trackId.toVariant());
    }
```
because otherwise this branch receives track load signals from samplers, for example when starting Mixxx.

Reply:
Thank you for this feedback! Just changed it 👍